crates/environ/src/builtin.rs

```diff
@@ -3,80 +3,88 @@
             check_free(vmctx: vmctx, addr: i32) -> i32;
-            /// Invoked before a load is executed.
+            // Invoked before a load is executed.
+            #[cfg(feature = "wmemcheck")]
             check_load(vmctx: vmctx, num_bytes: i32, addr: i32, offset: i32) -> i32;
-            /// Invoked before a store is executed.
+            // Invoked before a store is executed.
+            #[cfg(feature = "wmemcheck")]
             check_store(vmctx: vmctx, num_bytes: i32, addr: i32, offset: i32) -> i32;
-            /// Invoked after malloc is called.
+            // Invoked after malloc is called.
+            #[cfg(feature = "wmemcheck")]
             malloc_start(vmctx: vmctx);
-            /// Invoked after free is called.
+            // Invoked after free is called.
+            #[cfg(feature = "wmemcheck")]
             free_start(vmctx: vmctx);
-            /// Invoked when wasm stack pointer is updated.
+            // Invoked when wasm stack pointer is updated.
+            #[cfg(feature = "wmemcheck")]
             update_stack_pointer(vmctx: vmctx, value: i32);
-            /// Invoked before memory.grow is called.
+            // Invoked before memory.grow is called.
+            #[cfg(feature = "wmemcheck")]
             update_mem_size(vmctx: vmctx, num_bytes: i32);
 
-            /// Returns an index to drop a `VMExternRef`.
+            // Returns an index to drop a `VMExternRef`.
             #[cfg(feature = "gc")]
             drop_externref(vmctx: vmctx, val: pointer);
 
-            /// Returns an index to do a GC and then insert a `VMExternRef` into the
-            /// `VMExternRefActivationsTable`.
+            // Returns an index to do a GC and then insert a `VMExternRef` into the
+            // `VMExternRefActivationsTable`.
             #[cfg(feature = "gc")]
             activations_table_insert_with_gc(vmctx: vmctx, val: reference);
 
-            /// Returns an index for Wasm's `global.get` instruction for `externref`s.
+            // Returns an index for Wasm's `global.get` instruction for `externref`s.
             #[cfg(feature = "gc")]
             externref_global_get(vmctx: vmctx, global: i32) -> reference;
 
-            /// Returns an index for Wasm's `global.get` instruction for `externref`s.
+            // Returns an index for Wasm's `global.get` instruction for `externref`s.
             #[cfg(feature = "gc")]
             externref_global_set(vmctx: vmctx, global: i32, val: reference);
 
-            /// Returns an index for Wasm's `table.grow` instruction for `externref`s.
+            // Returns an index for Wasm's `table.grow` instruction for `externref`s.
             #[cfg(feature = "gc")]
             table_grow_externref(vmctx: vmctx, table: i32, delta: i32, init: reference) -> i32;
 
-            /// Returns an index for Wasm's `table.fill` instruction for `externref`s.
+            // Returns an index for Wasm's `table.fill` instruction for `externref`s.
```

Comment:
It seems nicer to `allow(unused_doc_comments)` in a few places than to lose the doc comments themselves, since they show up not only in rustdoc but also in editors using rust-analyzer/LSP and all that on hover and such.

Reply:
Good question! The answer actually relates to your below comment of why to define panicking stubs rather than omitting the functions entirely. I actually first went down the route of entirely omitting the libcalls from both numbering and runtime, but I stopped after implementing that and realized that it could create a confusing situation:

* One machine, with wasmtime features enabled, compiles a wasm module.
* Another machine, with minimal features enabled, loads this module. 

These two machines have different libcall numbering schemes without knowing it. I ended up realizing this would be a pretty confusing issue to run into, so I wanted to instead have all the libcalls always have the same number whether they're actually needed or not at runtime.

That then gave rise to needing to have something deal with `#[cfg]` and `#[cfg(not)]`. This is below in the trampoline where it uses `#[cfg(not(..))]` and that changes the pattern of the macro because it no longer matches all attributes but only `#[cfg]` attributes on each of these items.

More-or-less I couldn't find a nice and easy way to make a macro pattern that would separate doc comments and cfg attributes, so I figured the loss of some minor rustdoc wasn't the end of the world and I demoted these to normal comments instead of doc comments.

Reply:
Makes sense, reasonable tradeoff. Thanks for explaining!